### PR TITLE
config: Add S3 backed storage for computed block structures

### PR DESCRIPTION
**Pull request by blarghmatey**
### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4436

### Description (What does it do?)

Configures the edX course structures to be stored in S3 after being computed for better performance

### CODE REVIEW

**Review by asadali145 — CHANGES_REQUESTED** [review]
block_structure.storage_backing_for_cache is a WaffleSwitch, I don't think this will go with the flags.

We only wanted to test it for MITx Online but it is recommended to be used in Production so I am OK with enabling it for other apps.

For xPro we have added the storage configs but haven't enabled the Switch. If we cannot create a switch in ol-infra then we can do that in Django Admin.

**Review by feoh — APPROVED** [review]

File: src/ol_infrastructure/applications/edxapp/Pulumi.applications.edxapp.mitx-staging.CI.yaml [hunk 1]
```diff
@@ -5,6 +5,17 @@ config:
   aws:region: us-east-1
   consul:address: https://consul-mitx-staging-ci.odl.mit.edu
   edxapp:waffle_flags:
+  - ["block_structure.storage_backing_for_cache", "--create", "--everyone"]
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/ol_infrastructure/applications/edxapp/Pulumi.applications.edxapp.mitx-staging.CI.yaml:8]: Can you clarify the distinction between a waffle switch and waffle flag? We are using the manage.py CLI for setting these values.

[the remark this answers is not in the document]
Comment (asadali145) [reply src/ol_infrastructure/applications/edxapp/Pulumi.applications.edxapp.mitx-staging.CI.yaml:8]: @blarghmatey Waffle Flag and Waffle Switch are 2 different implementations and have different fields.
https://courses-qa.mitxonline.mit.edu/admin/waffle/flag/
https://courses-qa.mitxonline.mit.edu/admin/waffle/switch/

A switch is active or in-active globally while flags are active or inactive per request.
https://github.com/jazzband/django-waffle/blob/master/waffle/models.py#L132
https://github.com/jazzband/django-waffle/blob/master/waffle/models.py#L428

If we are using this [`waffle_flag`](https://github.com/jazzband/django-waffle/blob/master/waffle/management/commands/waffle_flag.py) command to create flags then there is another one to create switches https://github.com/jazzband/django-waffle/blob/master/waffle/management/commands/waffle_switch.py. [`waffle_switch`](https://github.com/jazzband/django-waffle/blob/master/waffle/management/commands/waffle_switch.py) has different set of arguments.

[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/ol_infrastructure/applications/edxapp/Pulumi.applications.edxapp.mitx-staging.CI.yaml:8]: Thank you for the clarification and references. Given that, you are correct that this is in the wrong location. We do not currently have anything in place to manage Waffle switches, but we should be able to do that with minimal effort. For now I'll remove this flag from the Pulumi stacks and rely on managing the switch manually until we get an automated mechanism in place.

[the remark this answers is not in the document]
Comment (asadali145) [reply src/ol_infrastructure/applications/edxapp/Pulumi.applications.edxapp.mitx-staging.CI.yaml]: Sure, Sounds good to me!

**Review by asadali145 — APPROVED** [review]
LGTM!

**Pull request merged**